datalad/support/gitrepo.py

```diff
@@ -1674,13 +1674,20 @@ def commit(self, msg=None, options=None, _datalad_msg=False, careless=True,
 
         orig_msg = msg
         if not msg:
-            msg = 'Recorded changes'
-            _datalad_msg = True
+            if '--amend' in options and '--no-edit' not in options:
+                # don't overwrite old commit message with our default message
+                # by default, but re-use old one. In other words: Make --no-edit
+                # the default:
+                options += ["--no-edit"]
```

Comment:
Shouldn't the `'--no-edit' not in options` check be moved within the `'--amend' in options` arm?  Otherwise, if `commit` is called directly with `--amend` _and_  `--no-edit`, the generic 'Recorded changes' message would be used.

```diff
diff --git a/datalad/support/tests/test_gitrepo.py b/datalad/support/tests/test_gitrepo.py
index 8deb1f1396..b62b58b870 100644
--- a/datalad/support/tests/test_gitrepo.py
+++ b/datalad/support/tests/test_gitrepo.py
@@ -335,6 +335,11 @@ def test_GitRepo_commit(path):
     assert_equal(len(list(gr.get_branch_commits_())), 2)
     assert_equal(gr.format_commit("%B").strip(), "amend message")

+    gr.commit(options=["--amend", "--no-edit"])
+    assert_repo_status(gr)
+    assert_equal(len(list(gr.get_branch_commits_())), 2)
+    assert_equal(gr.format_commit("%B").strip(), "amend message")
+
     # nothing to commit doesn't raise by default:
     gr.commit()
     # but does with careless=False:
```

```
$ python -m nose -vs datalad.support.tests.test_gitrepo:test_GitRepo_commit                                                                                   datalad.support.tests.test_gitrepo.test_GitRepo_commit ... FAIL
[...]
======================================================================
FAIL: datalad.support.tests.test_gitrepo.test_GitRepo_commit
----------------------------------------------------------------------
[...]
    assert_equal(gr.format_commit("%B").strip(), "amend message")
AssertionError: '[DATALAD] Recorded changes' != 'amend message'
- [DATALAD] Recorded changes
+ amend message
```

Reply:
Oops. Correct!